flow/project.py

```diff
@@ -686,6 +720,33 @@ def _show_template_help_and_exit(self, template_environment, context):
             filters='\n'.join(wrapper.wrap(', '.join(sorted(template_environment.filters))))))
         sys.exit(2)
 
+    @property
+    def hooks(self):
+        "Return a reference to the instance of :class:`.hooks.Hook` of this project."
+        return self._hooks
+
+    def _install_config_hooks(self):
+        try:
+            hooks_config = self._config['flow'].as_list('hooks')
+        except KeyError:
+            return  # no hooks configured
+        for entry in hooks_config:
+            try:
+                m = re.match(r'^(?P<path>[\w.]+?)(\((?P<constructor>.*)\))?$', entry)
```

Comment:
would it make sense to compile this regex (outside of the for loop) then use it for searches in the loop?

Reply:
No, the `re` package has been optimized to automatically cache those and this loop would only be executed a few times, not hundreds or thousands of times anyways.